packages/protvista-variation-adapter/src/protvista-variation-adapter.ts

```diff
@@ -49,37 +35,27 @@ export const transformData = (
 class ProtvistaVariationAdapter
   extends ProtvistaFeatureAdapter
   implements NightingaleElement {
-  static get is() {
+  static get is(): string {
```

Comment:
I believe this type is already defined in Nightingale element. So hopefully we don't need to write it here as it should be implied.
Ideally we'd want all the standard custom element methods to be the same to, no need to type them. But I'm not sure if there's a proper definition with TypeScript. I belive there is `CustomElementConstructor` that would just define the constructor. Maybe we need to have NightingaleElement extend a CustomElement (hopefully already defined somewhere, otherwise we'll have to type it), that would itself extend HTMLElement (which does exist for sure)

Reply:
As an aside, I think we need a place to define abstract classes used by multiple packages (`NightingaleElement` is currently defined in `data-loader`). There's quite a lot of "magic" methods in `protvista-track` and in the data adapters and having a common interface defined for these would be useful. What would be the best here...a package which is just exporting these shared type definitions?

In this specific case, `ProtvistaVariationAdapter` doesn't seem to pick up that the `is` getter corresponds to the readonly attribute in `NightingaleElement`. Actually looking at `DataLoader` it looks like you had to re-define the output type there too...

Reply:
Yeah, don't really remember what happened when I did it, I was just listing what I wished it would do. I guess we can re-work that when we have a look at Nightingale as a whole next month. I was thinking about an independent type package, or something like that. We'd need to have a look at what's the usual approach for shared resources with Lerna, but we might want to publish just the types themselves as their own package if people want to create their own "Nightingale-compatible" packages outside of our repo.

Reply:
That sounds like a good plan.